pinot-common/src/main/java/org/apache/pinot/common/function/scalar/StringFunctions.java

```diff
@@ -484,18 +484,19 @@ public static String[] split(String input, String delimiter) {
    */
   @ScalarFunction
   public static String[] split(String input, String delimiter, int limit) {
-    return StringUtils.split(input, delimiter, limit);
+    return StringUtils.splitByWholeSeparator(input, delimiter, limit);
   }
 
   /**
+   * TODO: Revisit if index should be zero-based
```

Comment:
What do you mean by `zero-based` here?

Reply:
See the discussion under #9554. Added more comments to make it more clear